src/app/venter/complaints-home/complaints-home.component.html

```diff
@@ -0,0 +1,152 @@
+                                    </div>
+                                    <div *ngIf="complaint.suggestions">
+                                        <div class="complaint-card-content"><b>Suggestions:</b></div>
+                                        <div>{{ complaint.suggestions }}</div>
+                                    </div>
+                                    <div *ngIf="complaint.location_details">
+                                        <div class="complaint-card-content"><b>Location Details:</b></div>
+                                        <div style="padding-top: 2%">{{ complaint.location_details }}</div>
+                                    </div>
+                                </div>
+                            </mat-card-content>
+                            <mat-card-footer>
+                                <div class="footer">
+                                    <div>
+                                        <div class="footer-button-upvote">
+                                            <button (click)="upVote(complaint.vote_count, complaint.id)"
+                                                mat-icon-button>
+                                                <mat-icon>arrow_upward</mat-icon>{{ complaint.users_up_voted.length }}
+                                            </button>
+                                        </div>
+                                        <button mat-icon-button class="footer-button-comment">
+                                            <mat-icon>insert_comment</mat-icon>{{ complaint.comments.length }}
+                                        </button>
+                                        <div [ngSwitch]="complaint.is_subscribed" class="footer-button-subscribe">
+                                            <div *ngSwitchCase="true">
+                                                <button (click)="subscribe(complaint.complaint_subscribed)"
+                                                    mat-icon-button>
+                                                    <mat-icon>notifications_active</mat-icon>
+                                                </button>
+                                            </div>
+                                            <div *ngSwitchCase="false">
+                                                <button (click)="subscribe(complaint.complaint_subscribed)"
+                                                    mat-icon-button>
+                                                    <mat-icon>notifications</mat-icon>
+                                                </button>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </mat-card-footer>
+                        </mat-card>
+                    </div>
+                </div>
+            </mat-tab>
```

Comment:
This is going to be crazy difficult to maintain. Consider splitting up into components.